src/components/modals/ViewCohortDetailsModal.tsx

```diff
@@ -0,0 +1,83 @@
+import { Controller, useForm } from "react-hook-form"
+import Button from "../ui/Button"
+import { Modal } from "@mui/material"
+import { Stage } from "../../utils/types"
+import Input from "../ui/Input"
+import dayjs from "dayjs"
+import UpdateStages from "../ui/UpdateStages"
+import { DatePicker } from "@mui/x-date-pickers"
+import CloseIcon from "../../assets/CloseIcon"
+
+interface ViewCohortDetailsModalProps {
+  onClose: () => void
+  cohort: {
+    _id: string
+    name: string
+    description: string
+    trainingStartDate: string
+    stages: Stage[]
+  }
+}
+
+export default function ViewCohortDetailsModal({
+  onClose,
+  cohort,
+}: ViewCohortDetailsModalProps) {
+  const {
+    register,
+    control,
+    formState: { errors },
+  } = useForm({
+    defaultValues: {
+      name: cohort?.name ?? "",
+      description: cohort?.description ?? "",
+      trainingStartDate: dayjs(cohort?.trainingStartDate),
+      stages: cohort?.stages ?? [],
+    },

```

Comment:
I suggest to not consider these properties that they might not be available because by the moment you click the view button to see the cohort details, it means that that cohort was there to begin with. So, ideally those details should already be there.
```suggestion
     defaultValues: {
      name: cohort.name,
      description: cohort.description,
      trainingStartDate: dayjs(cohort.trainingStartDate),
      stages: cohort.stages
    },
```